assignments/array-methods.js

```diff
@@ -53,31 +53,164 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
-// ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+
+
+
+/*********************************************************************************
+*                           CHALLENGE 2: Use .map()                              *
+**********************************************************************************/ 
+logHeader('CHALLENGE 2: Use .map()', 'Convert each first name into all caps and log the result:');
+
+/* 
+    The event director needs to have all the runner's first names converted to 
+    uppercase because the director BECAME DRUNK WITH POWER. Convert each first name 
+    into all caps and log the result
+*/
+
+let allCaps = runners.map(runner => runner.first_name.toUpperCase());
+
 console.log(allCaps); 
 
-// ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+
+
+
+/*********************************************************************************
+*                           CHALLENGE 3: Use .filter()                           *
+**********************************************************************************/ 
+logHeader('CHALLENGE 3: Use .filter()', 'Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result:');
+
+/*  
+    The large shirts won't be available for the event due to an ordering issue.  
+    Get a list of runners with large sized shirts so they can choose a different size. 
+    Return an array named largeShirts that contains information about the runners that have 
+    a shirt size of L and log the result
+*/
+
+let largeShirts = runners.filter(runner => runner.shirt_size === 'L');
+
 console.log(largeShirts);
 
-// ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+
+
+
+/*********************************************************************************
+*                           CHALLENGE 4: Use .reduce()                           *
+**********************************************************************************/ 
+logHeader('CHALLENGE 4: Use .reduce()', 'Add up all the donations into a ticketPriceTotal array and log the result:');
+
+/* 
+    The donations need to be tallied up and reported for tax purposes. Add up all 
+    the donations into a ticketPriceTotal array and log the result.
+*/
+
+let ticketPriceTotal = runners.map(runner => runner.donation).reduce((dono1, dono2) => dono1 + dono2);
```

Comment:
The semantics for the first parameter of your callback function could be improved a little. Remember, the first parameter accepts the accumulator, which in this case would be the running sum. So something like `totalDonations`, `runningTotal`, etc. may be more meaningful than `dono1`.